src/server/scripts/EasternKingdoms/Karazhan/boss_shade_of_aran.cpp

```diff
@@ -503,6 +529,31 @@ class boss_shade_of_aran : public CreatureScript
                     return;
                 }
         }
+
+        void MoveInLineOfSight(Unit* who) override
+        {
+            ScriptedAI::MoveInLineOfSight(who);
+
+            if (SeenAtiesh || me->IsInCombat() || me->GetDistance2d(who) > me->GetAttackDistance(who) + 10.0f)
+                return;
+
+            Player* player = who->ToPlayer();
+            if (!player)
+                return;
+
+            for (auto id : AtieshStaves)
```

Comment:
can be just for (auto const& i : {ITEM_ATIESH_MAGE, ITEM_ATIESH_WARLOCK, ITEM_ATIESH_PRIEST, ITEM_ATIESH_DRUID})   and it'll take less lines of code - faster to write it atleast, also'll not cause warnings and other shit